localstack/testing/pytest/marker_report.py

```diff
@@ -0,0 +1,158 @@
+import dataclasses
+import datetime
+import json
+import os.path
+import time
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import pytest
+import requests
+
+if TYPE_CHECKING:
+    from _pytest.config import Config, PytestPluginManager
+    from _pytest.config.argparsing import Parser
+
+
+@dataclasses.dataclass
+class MarkerReportEntry:
+    node_id: str
+    file_path: str
+    markers: "list[str]"
+
+
+@dataclasses.dataclass
+class TinybirdReportRow:
+    timestamp: str
+    node_id: str
+    project_name: str
+    # code_owners: str  # comma separated
+    file_path: str  # TODO: recreate data source at some point to remove this?
+    service: str
+    markers: str  # comma separated list
+    aws_marker: str  # TODO: this is a bit redundant but easier for now to process
+    commit_sha: str
+
+
+@dataclasses.dataclass
+class MarkerReport:
+    prefix_filter: str
+    entries: "list[MarkerReportEntry]" = dataclasses.field(default_factory=list)
+    aggregated_report: "dict[str, int]" = dataclasses.field(default_factory=dict)
+
+    def create_aggregated_report(self):
+        for entry in self.entries:
+            for marker in entry.markers:
+                self.aggregated_report.setdefault(marker, 0)
+                self.aggregated_report[marker] += 1
+
+
+@pytest.hookimpl
+def pytest_addoption(parser: "Parser", pluginmanager: "PytestPluginManager"):
```

Comment:
nit: typo? or: add_option?

Reply:
Its just how pytest names this hook. Unfortunately not really something we have any influence over :sweat_smile: